src/App.tsx

```diff
@@ -1,14 +1,61 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [todosStatus, setTodosStatus] = useState<boolean | null>(null);
+  const [todosQuery, setTodosQuery] = useState('');
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+
+  useEffect(() => {
+    getTodos().then(setTodos);
+  }, []);
+
+  const setterOfSelectedTodo = (pickedTodo: Todo | null) => {
+    setSelectedTodo(pickedTodo);
+  };
+
+  const filterByStatus = (option: string) => {
+    if (option === 'all') {
+      setTodosStatus(null);
+    }
+
+    if (option === 'active') {
+      setTodosStatus(false);
+    }
+
+    if (option === 'completed') {
+      setTodosStatus(true);
+    }
+  };
+
+  const filterByQuery = (query: string) => {
+    setTodosQuery(query);
+  };
+
+  let filteredTodos = [...todos];
+
+  if (todosStatus !== null) {
+    filteredTodos = filteredTodos
+      .filter(todoStatus => todoStatus.completed === todosStatus);
+  }
+
+  if (todosQuery.trim() !== '') {
+    const lowerQuery = todosQuery.toLocaleLowerCase().trim();
+
+    filteredTodos = filteredTodos
+      .filter(todoQuery => todoQuery.title.toLocaleLowerCase().trim().includes(lowerQuery));
```

Comment:
i guess you don't have to trim todos title here, no strong opinion tho

Reply:
> i guess you don't have to trim todos title here, no strong opinion tho

lets leave it, i like to add some spaces before entering smth ;)